internal/provision/devices.go

```diff
@@ -26,79 +25,39 @@ import (
 	"github.com/google/uuid"
 	"github.com/hashicorp/go-multierror"
 	"gopkg.in/yaml.v3"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
 
 	"github.com/edgexfoundry/device-sdk-go/v3/internal/cache"
 	"github.com/edgexfoundry/device-sdk-go/v3/internal/container"
 )
 
 func LoadDevices(path string, dic *di.Container) errors.EdgeX {
+	var addDevicesReq []requests.AddDeviceRequest
+	var edgexErr errors.EdgeX
 	if path == "" {
 		return nil
 	}
 
-	absPath, err := filepath.Abs(path)
-	if err != nil {
-		return errors.NewCommonEdgeX(errors.KindServerError, "failed to create absolute path", err)
-	}
-
-	files, err := os.ReadDir(absPath)
-	if err != nil {
-		return errors.NewCommonEdgeX(errors.KindServerError, "failed to read directory", err)
-	}
-
-	if len(files) == 0 {
-		return nil
-	}
-
 	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
-	lc.Infof("Loading pre-defined devices from %s(%d files found)", absPath, len(files))
-
-	var addDevicesReq []requests.AddDeviceRequest
 	serviceName := container.DeviceServiceFrom(dic.Get).Name
-	for _, file := range files {
-		var devices []dtos.Device
-		fullPath := filepath.Join(absPath, file.Name())
-		if strings.HasSuffix(fullPath, yamlExt) || strings.HasSuffix(fullPath, ymlExt) {
-			content, err := os.ReadFile(fullPath)
-			if err != nil {
-				lc.Errorf("Failed to read %s: %v", fullPath, err)
-				continue
-			}
-			d := struct {
-				DeviceList []dtos.Device `yaml:"deviceList"`
-			}{}
-			err = yaml.Unmarshal(content, &d)
-			if err != nil {
-				lc.Errorf("Failed to YAML decode %s: %v", fullPath, err)
-				continue
-			}
-			devices = d.DeviceList
-		} else if strings.HasSuffix(fullPath, ".json") {
-			content, err := os.ReadFile(fullPath)
-			if err != nil {
-				lc.Errorf("Failed to read %s: %v", fullPath, err)
-				continue
-			}
-			err = json.Unmarshal(content, &devices)
-			if err != nil {
-				lc.Errorf("Failed to JSON decode %s: %v", fullPath, err)
-				continue
-			}
-		} else {
-			continue
+	parsedUrl, err := url.Parse(path)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindServerError, "failed to parse devices path as a url", err)
```

Comment:
URI should be all caps in all the log messages and be consistent with use of URI rather than URL
```suggestion
		return errors.NewCommonEdgeX(errors.KindServerError, "failed to parse devices path as a URI", err)
```